signer/sign_state.go

```diff
@@ -107,7 +107,11 @@ func (signState *SignState) GetFromCache(hrs HRSKey, lock *sync.Mutex) (HRSKey,
 	return latestBlock, nil
 }
 
-func (signState *SignState) Save(ssc SignStateConsensus, lock *sync.Mutex, async bool) error {
+func (signState *SignState) Save(
+	ssc SignStateConsensus,
+	lock *sync.Mutex,
```

Comment:
Maybe outside the scope of this PR, but maybe not since you already had to touch all the occurrences of this method:

passing a mutex to a function, just to have it lock and unlock, is an awkward use of a parameter. The caller should manage the lock themselves.

Reply:
good call, updated.